saleor/discount/utils.py

```diff
@@ -677,6 +840,107 @@ def _create_or_update_checkout_discount(
             ]
         )
 
+    _delete_gift_line(lines_info)
+
+
+def _delete_gift_line(lines_info: Iterable["CheckoutLineInfo"]):
+    gift_line_infos = [line for line in lines_info if line.line.is_gift]
+    if gift_line_infos:
+        gift_line_info = gift_line_infos[0]
```

Comment:
Shouldn't we handle here multiple gitfts? I mean it doesn't look like a massive work, but we could miss this logic when extending the gift limit 😅

Reply:
Done